Fix sourcemaps for AOT

tbosch opened the pull request:
Includes https://github.com/angular/angular/pull/15191

tbosch committed:
fix(tsc-wrapped): emit js files in all cases

tbosch committed:
fix(compiler): make sourcemaps work in AOT mode

Inlcuded fixes:
- include preamble in generated source map
- always add a mapping for line/col 0 so that the
  generated sourcemap is not sparse
- use a uniue sourceUrl for inline templates even
  in the AOT case

alexeagle commented on tools/@angular/tsc-wrapped/src/main.ts:
```diff
@@ -117,7 +117,11 @@ export function main(
       let definitionsHost: ts.CompilerHost = tsickleCompilerHost;
       if (!ngOptions.skipMetadataEmit) {
-        definitionsHost = new MetadataWriterHost(tsickleCompilerHost, ngOptions);
+        // if tsickle is not not used for emitting, but we do use the MetadataWriteHost,
```
typo metadatawrite*r*host

alexeagle reviewed (commented):
getting gross, but let's hope I get the tsc_wrapped refactoring going soon.

LucasSloan committed:
build(tsc-wrapped): update tsickle to version 0.21.6

Update tsickle to version 0.21.6 which fixes a bug where input source maps which specified filenames differently than the names supplied to tsc didn't get composed with tsc's source maps.  Also adds a test that the bug was fixed.

alexeagle reviewed (approved):
only approving for the first commit
